tests/pytests/unit/utils/test_network.py

```diff
@@ -1,8 +1,1547 @@
+            "inet": [{"address": "127.0.0.1", "netmask": "255.0.0.0"}],
+            "inet6": [
+                {"address": "fe80::1", "prefixlen": "64", "scope": "0x8"},
+                {"address": "::1", "prefixlen": "128", "scope": None},
+            ],
+            "up": True,
+        },
+        "plip0": {"up": False},
+        "tun0": {
+            "inet": [{"address": "10.12.0.1", "netmask": "255.255.255.255"}],
+            "up": True,
+        },
+    }
+
+
+def test_sanitize_host_ip():
+    ret = network.sanitize_host("10.1./2.$3")
+    assert ret == "10.1.2.3"
+
+
+def test_sanitize_host_name():
+    """
+    Should not remove the underscore
+    """
+    ret = network.sanitize_host("foo_bar")
+    assert ret == "foo_bar"
+
+
+def test_host_to_ips():
+    """
+    NOTE: When this test fails it's usually because the IP address has
+    changed. In these cases, we just need to update the IP address in the
+    assertion.
+    """
+
+    # pylint doesn't like the }[host] below, disable typecheck
+    # pylint: disable=all
+    def getaddrinfo_side_effect(host, *args):
+        try:
+            return {
+                "github.com": [
+                    (2, 1, 6, "", ("192.30.255.112", 0)),
+                    (2, 1, 6, "", ("192.30.255.113", 0)),
+                ],
+                "ipv6host.foo": [
+                    (socket.AF_INET6, 1, 6, "", ("2001:a71::1", 0, 0, 0)),
+                ],
+            }[host]
+        except KeyError:
+            raise socket.gaierror(-2, "Name or service not known")
+
+    # pylint: enable=all
+
+    getaddrinfo_mock = MagicMock(side_effect=getaddrinfo_side_effect)
+    with patch.object(socket, "getaddrinfo", getaddrinfo_mock):
+        # Test host that can be resolved
+        ret = network.host_to_ips("github.com")
+        assert ret == ["192.30.255.112", "192.30.255.113"]
+
+        # Test ipv6
+        ret = network.host_to_ips("ipv6host.foo")
+        assert ret == ["2001:a71::1"]
+        # Test host that can't be resolved
+        ret = network.host_to_ips("someothersite.com")
+        assert ret is None
+
+
+def test_generate_minion_id():
+    assert network.generate_minion_id()
+
+
+def test__generate_minion_id_with_unicode_in_etc_hosts():
+    """
+    Test that unicode in /etc/hosts doesn't raise an error when
+    _generate_minion_id() helper is called to gather the hosts.
+    """
+    content = textwrap.dedent(
+        """\
+    # 以下为主机名解析
+    ## ccc
+    127.0.0.1       localhost thisismyhostname     # 本机
+    """
     )
+    fopen_mock = mock_open(read_data={"/etc/hosts": content})
+    with patch("salt.utils.files.fopen", fopen_mock):
+        assert "thisismyhostname" in network._generate_minion_id()
+
+
+def test_is_ip():
+    assert network.is_ip("10.10.0.3")
+    assert not network.is_ip("0.9.800.1000")
+    # Check 16-char-long unicode string
+    # https://github.com/saltstack/salt/issues/51258
+    assert not network.is_ipv6("sixteen-char-str")
```

Comment:
```suggestion
```
Doesn't test `is_ip()` and this particular assertion is tested on `test_is_ipv4` and `test_is_ipv6`.

Reply:
updated it to is_ip